internal/renderer/cmark/cmark.go

```diff
@@ -314,11 +314,21 @@ func (r *renderer) Render(doc ast.Node, source []byte) ([]byte, error) {
 			}
 
 		case ast.KindEmphasis:
-			n := node.(*ast.Emphasis)
-			mark := []byte{'*'}
-			if n.Level > 1 {
-				mark = []byte{'*', '*'}
+			var p ast.Node
+			for p = node.Parent(); p.Type() == ast.TypeInline; {
+				p = p.Parent()
+			}
+			underscores := 0
+			for i := 0; i < p.Lines().Len(); i++ {
+				line := p.Lines().At(i)
+				underscores += bytes.Count(line.Value(source), []byte{'_'})
 			}
+			mark := []byte{'_'}
+			if underscores == 0 {
+				mark = []byte{'*'}
+			}
+			n := node.(*ast.Emphasis)
+			mark = bytes.Repeat(mark, n.Level)
```

Comment:
This will fail on the following:

```markdown
[[ *TOC* _TOC_ ]]
```

Since it will read the first asterisk, and assume they are both asterisks.

I looked into goldmark's custom parsing capabilities quite a bit, but I wasn't able to find anything super promising. I thought you could wrap the inline parser and set an attribute on the node, but that didn't seem possible. And the AST transform system doesn't seem flexible enough. Though it still may be possible

Reply:
As discussed in person, mixed usage like that is rather unlikely, and we're ok unifying one or the other.